### PR TITLE
Configurable cache headers

### PR DESCRIPTION
**Pull request by jphenow**
Sorry, I didn't see a contribution guide, but this looked like a fun little issue to attempt.

I didn't see much for tests around this kind of thing so also happy to look into those with a little guidance.

For #144

### CODE REVIEW

File: v8env/src/fly/cache/index.ts [hunk 1]
```diff
@@ -20,9 +20,10 @@
 /** */
 declare var bridge: any
 export interface CacheSetOptions {
-  ttl?: number,
-  tags?: string[],
-  onlyIfEmpty?: boolean
+  ttl?: number;
+  tags?: string[];
+  onlyIfEmpty?: boolean;
+  skipCacheHeaders?: string[];
```
Comment (mrkurt) [inline v8env/src/fly/cache/index.ts]: The `skipCacheHeaders` only applies to the response cache, so probably better to make a new `ResponseCacheSetOptions` for this purpose.

**Comment by jphenow** [comment]
Hmm running locally I get

```
ERROR in ./v8env/lib/streams.js
Module not found: Error: Can't resolve 'whatwg-streams/reference-implementation/lib/transform-stream' in '/Users/jonphenow/Code/fly/v8env/lib'
 @ ./v8env/lib/streams.js 10:481-552
 @ ./v8env/lib/index.js
```

Is there something obvious I'm missing?

**Comment by mrkurt** [comment]
Ah, try `git submodule update --init --recursive`.

**Comment by jphenow** [comment]
🎉 That got me new results 😄 

```
Error: Cannot find module '/Users/jonphenow/Code/fly/lib/cmd/index.js'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
error Command failed with exit code 1.
```

**Comment by mrkurt** [comment]
`npm run build` should get past that one.

**Comment by jphenow** [comment]
Oh yep there we go - I was thinking that was run as a prerequisite task to the test run - thanks!

File: v8env/src/fly/cache/response.ts [hunk 2]
```diff
@@ -126,10 +130,18 @@ export async function set(key: string, resp: Response, options?: CacheSetOptions
     resp.headers.set("etag", etag)
   }
 
-  for (const h of goodHeaders) {
-    const v = resp.headers.getAll(h);
-    if (v && v.length > 0) {
-      meta.headers[h] = v.join(', ');
+  const skipHeaderSet = new Set(skipHeaderOption);
+  for(const headerSet of resp.headers as any) {
```
[the remark this answers is not in the document]
Comment (mrkurt) [reply v8env/src/fly/cache/response.ts]: The Fly `Response` object is actually an implementation of the Web API, so the one from `lib.dom` is "right". When they diverge it means we have a bug. :D

I'll have a look in an hour or two and see if there's a way to make it more idiomatic, the `any` is probably fine though!

[the remark this answers is not in the document]
Comment (mrkurt) [reply v8env/src/fly/cache/response.ts]: Ok I think this is the best it's gonna get until #162. When we have a complete headers implementation we can clean this up a little bit!

**Pull request merged**